### PR TITLE
Logocert api

### PR DESCRIPTION
**Pull request by lrbom**
Please review my changes.

- All web driver functionalities have been replaced by GCP or Privet API calls

- Added GCP Update and Register interface

- Added various Privet interfaces for local printing

- Replaced blind waits with event-driven, polling based waits (device on/off, registration status, print job status) 

- Improved console output

- Bumped version to 2.0

### CODE REVIEW

File: testcert.py [hunk 1]
```diff
@@ -1722,232 +1934,233 @@ def testDeviceCancelRegistration(self):
...
+          self.assertTrue(success)
+        except AssertionError:
+          notes = 'User1 failed to register.'
+          self.LogTest(test_id, test_name, 'Failed', notes)
+          _device.CancelRegistration()
+          raise
+        else:
+          print 'Waiting up to 2 minutes to complete the registration.'
+          success = waitForAdvertisementRegistrationStatus(_device.name, True, 120)
+          try:
```
Comment (bryangl) [inline testcert.py]: Why is an exception used here to test success?

File: testcert.py [hunk 2]
```diff
@@ -1957,310 +2170,297 @@ def testPrinterOffSendGoodbyePacket(self):
...
 
     if failed:
+      notes = 'Printer did not send goodbye packet when powered off.'
       self.LogTest(test_id, test_name, 'Failed', notes)
+      raise
     else:
+      notes = 'Printer sent goodbye packet when powered off.'
       self.LogTest(test_id, test_name, 'Passed', notes)
 
+    PromptUserAction('Power on the printer and wait...')
```
Comment (bryangl) [inline testcert.py]: It might be nice for the user to know how long the script is about to (potentially) wait for.

File: testcert.py [hunk 3]
```diff
@@ -108,83 +128,164 @@ def _ParseArgs():
...
 
-def tearDownModule():
-  chromedriver.CloseChrome()
+def LogTestSuite(name):
+  """Log a test result.
+
+  Args:
+    name: string, name of the testsuite that is logging.
+  """
+  print '=============================================================================================================='
```
Comment (lrbom) [inline testcert.py]: This is truncated to be under 80 characters.

File: testcert.py [hunk 4]
```diff
@@ -108,83 +128,164 @@ def _ParseArgs():
...
+          properties = v['info'].properties
+          return properties['id'] and 'online' in properties['cs'].lower()
+  return None
+
+def waitForAdvertisementRegistrationStatus(name, is_wait_for_reg, timeout):
+  """Wait for the device to privet advertise as registered or unregistered
+
+      Args:
+        name: string, device status to wait on
+        is_wait_for_reg: boolean, True for registered status, False for unregistered
```
Comment (lrbom) [inline testcert.py]: I agree that an Enum would make the calling code a lot clearer. But I like the fact that a boolean forces the input to be binary - either true or false; whereas I would have to potentially handle unexpected Enum inputs from the caller if for whatever reason the another field is added to the Enum in the future. Also, I would either have to create a module level Enum in testcert.py or jam it into _config.py, both solutions aren't appealing to me. Thoughts?

File: testcert.py [hunk 5]
```diff
@@ -212,44 +313,106 @@ def GetNewTokens():
...
+
+      Args:
+        pwg_path: string, destination file path of the pwg_raster image
+        img_path: string, src file path of the image to convert from
+      """
+
+  #
+  cjt = CloudJobTicket(_device.details['gcpVersion'], _device.cdd['caps'])
+
+  print '%s not found. Likely that this is the first time LocalPrinting suite is run.' % (pwg_path)
```
Comment (lrbom) [inline testcert.py]: Good catch. I've moved this print statement to the calling code.

File: testcert.py [hunk 6]
```diff
@@ -837,412 +1001,411 @@ def testDeviceRegistrationInvalidUserAuthToken(self):
...
     try:
-      self.assertIn('Serial Number', device.details)
+      self.assertIn('uuid', _device.details)
     except AssertionError:
       notes = 'Serial number not found in device details.'
       self.LogTest(test_id, test_name, 'Failed', notes)
       raise
     try:
-      self.assertGreaterEqual(len(device.details['Serial Number']), 1)
+      self.assertGreaterEqual(len(_device.details['uuid']), 1)
```
Comment (lrbom) [inline testcert.py]: That's good point. UUID is the field that is returned via the /printer GCP interface thus the code's been changed to use uuid as the key instead. I changed the test name to refer to UUID and added comments explaining its equivalence to SN.

File: testcert.py (hunk 4 above)
Comment (kdlucas) [inline testcert.py]: I like boolean here. Probably less susceptible to breakage.

**Comment by vamshikk** [comment]
Completed.

**Pull request closed without merging**